source/agora/node/Registry.d

```diff
@@ -677,8 +698,145 @@ private struct ZoneData
     /// Whether we are authoritative or we're just caching
     public bool authoritative;
 
-    /// The content of the zone
-    public TypedPayload[PublicKey] map;
+    /// Database to store data
+    private ManagedDatabase db;
+
+    /***************************************************************************
+
+         Params:
+           cache_db = Database instance
+
+    ***************************************************************************/
+
+    public this (ManagedDatabase cache_db)
+    {
+        this.db = cache_db;
+
+        this.db.execute("CREATE TABLE IF NOT EXISTS registry_"~ TemplateTableName ~ "_signature " ~
+            "(pubkey TEXT, signature TEXT NOT NULL, sequence INTEGER NOT NULL, " ~
+            "PRIMARY KEY(pubkey))");
+
+        this.db.execute("CREATE TABLE IF NOT EXISTS registry_"~ TemplateTableName ~ "_addresses " ~
+            "(pubkey TEXT, address TEXT NOT NULL, type INTEGER NOT NULL, " ~
+            "FOREIGN KEY(pubkey) REFERENCES registry_"~ TemplateTableName ~ "_signature(pubkey) ON DELETE CASCADE, " ~
+            "PRIMARY KEY(pubkey, address))");
+    }
+
+    /***************************************************************************
+
+         Returns:
+           Total number of name registries
+
+    ***************************************************************************/
+
+    public ulong count () @trusted
+    {
+        auto results = this.db.execute("SELECT COUNT(*) FROM registry_"~ TemplateTableName ~ "_signature");
+        if (!results.empty)
+        {
+            return results.oneValue!(ulong);
+        }
+
+        return 0;
+    }
+
+    /***************************************************************************
+
+         Returns:
+           All registries in persistent storage
+
+    ***************************************************************************/
+
+    public ResultRange getRegistries () @trusted
```

Comment:
I see now why this is needed. It would probably be better to implement an `opApply` then:
https://github.com/bosagora/agora/blob/23f48f8dd0b4f6d1999e2b735a0bfb94cbe4ce47/source/agora/consensus/pool/Transaction.d#L279-L301